Orange/widgets/utils/distmatrixmodel.py

```diff
@@ -0,0 +1,120 @@
+        self.colors: Optional[np.ndarray] = None
+        self.__header_data: Dict[Any, Optional[LabelData]] = {
+            Qt.Horizontal: LabelData(),
+            Qt.Vertical: LabelData()}
+        self.__zero_diag: bool = True
+        self.__span: Optional[float] = None
+
+        self.__header_font = QFont()
+        self.__header_font.setBold(True)
+
+    def set_data(self, distances):
+        self.beginResetModel()
+        self.distances = distances
+        self.__header_data = dict.fromkeys(self.__header_data, LabelData())
+        if distances is None:
+            self.__span = self.colors = None
+            return
+        self.__span = span = np.max(distances)
+
+        self.colors = \
+            (distances * (170 / span if span > 1e-10 else 0)).astype(int)
+        self.__zero_diag = \
+            distances.is_symmetric() and np.allclose(np.diag(distances), 0)
+        self.endResetModel()
+
+    def set_labels(self, orientation, labels: Optional[List[str]],
+                   colors: Optional[np.ndarray] = None):
+        self.__header_data[orientation] = LabelData(labels, colors)
+        rc, cc = self.rowCount() - 1, self.columnCount() - 1
+        self.headerDataChanged.emit(
+            orientation, 0, rc if orientation == Qt.Vertical else cc)
+        self.dataChanged.emit(self.index(0, 0), self.index(rc, cc))
+
+    def rowCount(self, parent=None):
+        if parent and parent.isValid() or self.distances is None:
+            return 0
+        return self.distances.shape[0]
+
+    def columnCount(self, parent=None):
+        if parent and parent.isValid() or self.distances is None:
+            return 0
+        return self.distances.shape[1]
+
+    def data(self, index, role=Qt.DisplayRole):
+        if role == Qt.TextAlignmentRole:
+            return Qt.AlignCenter | Qt.AlignVCenter
+        if self.distances is None:
+            return None
+
+        row, col = index.row(), index.column()
+        if role == Qt.DisplayRole and not (self.__zero_diag and row == col):
+            return float(self.distances[row, col])
+        if role == Qt.BackgroundRole:
+            return QBrush(QColor.fromHsv(120, self.colors[row, col], 255))
+        if role == Qt.ForegroundRole:
+            return QColor(Qt.black)  # the background is light-ish
+        if role == FixedFormatNumericColumnDelegate.ColumnDataSpanRole:
+            return 0., self.__span
+        return None
+
+    def headerData(self, ind, orientation, role):
+        if role == Qt.FontRole:
+            return self.__header_font
+
+        __header_data = self.__header_data[orientation]
+        if role == Qt.DisplayRole:
+            if __header_data.labels is not None \
+                    and ind < len(__header_data.labels):
+                return __header_data.labels[ind]
+
+        colors = self.__header_data[orientation].colors
+        if colors is not None:
+            color = colors[ind].lighter(180)
```

Comment:
The background color is too light (`Iris-setosa` used to be blue):

![image](https://user-images.githubusercontent.com/11465003/211263493-dc91c881-aa87-45f4-9c66-00614b24ef4d.png)